testsuite/tests/libratest/smoke_test.rs

```diff
@@ -2045,3 +2050,219 @@ fn parse_waypoint(db_bootstrapper_output: &str) -> Waypoint {
+        }
+        println!("Backup storage state: {}", state);
+        std::thread::sleep(Duration::from_secs(1));
+    }
+
+    bail!("Failed to create backup.");
+}
+
+fn db_backup(backup_service_port: u16, target_epoch: u64, target_version: Version) -> TempPath {
+    let now = Instant::now();
+    let bin_path = workspace_builder::get_bin("db-backup");
+    let metadata_cache_path1 = TempPath::new();
+    let metadata_cache_path2 = TempPath::new();
+    let backup_path = TempPath::new();
+
+    metadata_cache_path1.create_as_dir().unwrap();
+    metadata_cache_path2.create_as_dir().unwrap();
+    backup_path.create_as_dir().unwrap();
+
+    // spawn the backup coordinator
+    let mut backup_coordinator = Command::new(bin_path.as_path())
+        .current_dir(workspace_root())
+        .args(&[
+            "coordinator",
+            "run",
+            "--backup-service-address",
+            &format!("http://localhost:{}", backup_service_port),
+            "--transaction-batch-size",
+            "20",
+            "--state-snapshot-interval",
+            "50",
+            "--metadata-cache-dir",
+            metadata_cache_path1.path().to_str().unwrap(),
+            "local-fs",
+            "--dir",
+            backup_path.path().to_str().unwrap(),
+        ])
+        .spawn()
+        .unwrap();
+
+    // watch the backup storage, wait for it to reach target epoch and version
+    let wait_res = wait_for_backups(
+        target_epoch,
+        target_version,
+        now,
+        bin_path,
+        &metadata_cache_path2,
+        &backup_path,
+    );
+    backup_coordinator.kill().unwrap();
+    wait_res.unwrap();
+    backup_path
+}
+
+fn db_restore(backup_path: &Path, db_path: &Path) {
+    let now = Instant::now();
+    let bin_path = workspace_builder::get_bin("db-restore");
+    let metadata_cache_path = TempPath::new();
+
+    metadata_cache_path.create_as_dir().unwrap();
+
+    let output = Command::new(bin_path.as_path())
+        .current_dir(workspace_root())
+        .args(&[
+            "--target-db-dir",
+            db_path.to_str().unwrap(),
+            "auto",
+            "--metadata-cache-dir",
+            metadata_cache_path.path().to_str().unwrap(),
+            "local-fs",
+            "--dir",
+            backup_path.to_str().unwrap(),
+        ])
+        .output()
+        .unwrap();
+    if !output.status.success() {
+        panic!("db-restore failed, output: {:?}", output);
+    }
+    println!("Backup restored in {} seconds.", now.elapsed().as_secs());
+}
+
+fn transfer_continuously(
+    mut client: ClientProxy,
+    balance0: f64,
+    balance1: f64,
+    quit: Arc<AtomicBool>,
+) -> f64 {
+    let now = Instant::now();
+    let mut transferred = 0f64;
+    while !quit.load(Ordering::Relaxed) {
+        client
+            .transfer_coins(&["tb", "0", "1", "1", "Coin1"], true)
+            .unwrap();
+        transferred += 1.0;
+
+        assert!(compare_balances(
```

Comment:
Does `transfer_coin` take effect immediately?

Reply:
yea, the last parameter is "is_blocking" (think the cli)